client/src/components/Focusedzone/Focusedzone.js

```diff
@@ -0,0 +1,91 @@
+/* global document */
+import React, { Component, PropTypes } from 'react';
+import ReactDOM from 'react-dom';
+
+/**
+ * Wrapper component that fires a handler (`onClickOut`) when a user clicks
+ * outside of the component.
+ *
+ *
+ * Example:
+ *
+ * ```jsx
+ *    class TestComponent extends React.Component {
+ *      ...
+ *
+ *      handleClickOut() {
+ *        alert('User clicked outside of TestComponent.');
+ *      }
+ *
+ *      render() {
+ *        return (
+ *          <Focusedzone onClickOut={this.handleClickOut}>
+ *            <div>... </div>
+ *          </Focusedzone>
+ *        );
+ *       }
+ *    < /Focusedzone>
+ *  ```
+ */
+
+class Focusedzone extends Component {
+
+  constructor(props) {
+    super(props);
+
+    // This will be set to `true` in `handleElementClick` handler when user
+    // clicked inside the component's DOM and reset back to `false` in
+    // `handleDocumentClick`. This component knows when user clicked outside of
+    // its DOM by checking if `wasClicked` is false in `handleDocumentClick`.
+    // All of this works because of the order of event bubbling.
+    this.wasClicked = false;
+
+    this.handleElementClick = this.handleElementClick.bind(this);
+    this.handleDocumentClick = this.handleDocumentClick.bind(this);
+  }
+
+  componentDidMount() {
+    const el = ReactDOM.findDOMNode(this);
+
+    // Click event handlers on `el` and `document` will be both
+    // triggered when user clicked inside the compoent's DOM. The
+    // handler on `el` is triggered before the one on `document` based on event
+    // bubbling.
+    el.addEventListener('click', this.handleElementClick);
+    document.addEventListener('click', this.handleDocumentClick);
+  }
+
+  componentWillUnmount() {
+    const el = ReactDOM.findDOMNode(this);
+    el.removeEventHandler('click', this.handleElementClick);
+    el.removeEventHandler('click', this.handleDocumentClick);
+  }
+
+  handleElementClick() {
+    this.wasClicked = true;
+  }
+
+  handleDocumentClick() {
+    if (!this.wasClicked) {
+      this.props.onClickOut();
+    } else {
```

Comment:
remove the `else` :) otherwise if `wasClicked` has been set true once, it won't reset afterwards